contractcourt/commit_sweep_resolver_test.go

```diff
@@ -0,0 +1,187 @@
+	}
+
+	sweeper := newMockSweeper()
+
+	checkPointChan := make(chan struct{}, 1)
+
+	chainCfg := ChannelArbitratorConfig{
+		ChainArbitratorConfig: ChainArbitratorConfig{
+			Notifier: notifier,
+			Sweeper:  sweeper,
+		},
+	}
+
+	cfg := ResolverConfig{
+		ChannelArbitratorConfig: chainCfg,
+		Checkpoint: func(_ ContractResolver) error {
+			checkPointChan <- struct{}{}
+			return nil
+		},
+	}
+
+	resolver := newCommitSweepResolver(
+		*resolution, 0, wire.OutPoint{}, cfg,
+	)
+
+	return &commitSweepResolverTestContext{
+		resolver: resolver,
+		notifier: notifier,
+		sweeper:  sweeper,
+		t:        t,
+	}
+}
+
+func (i *commitSweepResolverTestContext) resolve() {
+	// Start resolver.
+	i.resolverResultChan = make(chan resolveResult, 1)
+	go func() {
+		nextResolver, err := i.resolver.Resolve()
+		i.resolverResultChan <- resolveResult{
+			nextResolver: nextResolver,
+			err:          err,
+		}
+	}()
+}
+
+func (i *commitSweepResolverTestContext) notifyEpoch(height int32) {
+	i.notifier.epochChan <- &chainntnfs.BlockEpoch{
+		Height: height,
+	}
+}
+
+func (i *commitSweepResolverTestContext) waitForResult() {
+	i.t.Helper()
+
+	result := <-i.resolverResultChan
+	if result.err != nil {
+		i.t.Fatal(result.err)
+	}
+
+	if result.nextResolver != nil {
+		i.t.Fatal("expected no next resolver")
+	}
+}
+
+type mockSweeper struct {
+	sweptInputs chan input.Input
+}
+
+func newMockSweeper() *mockSweeper {
+	return &mockSweeper{
+		sweptInputs: make(chan input.Input),
+	}
+}
+
+func (s *mockSweeper) SweepInput(input input.Input,
+	feePreference sweep.FeePreference) (chan sweep.Result, error) {
+
+	s.sweptInputs <- input
+
+	result := make(chan sweep.Result, 1)
+	result <- sweep.Result{
+		Tx: &wire.MsgTx{},
+	}
+	return result, nil
+}
+
+func (s *mockSweeper) CreateSweepTx(inputs []input.Input, feePref sweep.FeePreference,
+	currentBlockHeight uint32) (*wire.MsgTx, error) {
+
+	return nil, nil
+}
+
+var _ UtxoSweeper = &mockSweeper{}
+
+// TestCommitSweepResolverNoDelay tests resolution of a direct commitment output
+// unencumbered by a time lock.
+func TestCommitSweepResolverNoDelay(t *testing.T) {
+	t.Parallel()
+	defer timeout(t)()
+
+	res := lnwallet.CommitOutputResolution{
+		SelfOutputSignDesc: input.SignDescriptor{
+			Output: &wire.TxOut{
+				Value: 100,
+			},
+		},
+	}
+
+	ctx := newCommitSweepResolverTestContext(t, &res)
+	ctx.resolve()
+
+	ctx.notifier.confChan <- &chainntnfs.TxConfirmation{}
+
+	// No csv delay, so the input should be swept immediately.
+	<-ctx.sweeper.sweptInputs
+
+	ctx.waitForResult()
+}
+
+// TestCommitSweepResolverDelay tests resolution of a direct commitment output
+// that is encumbered by a time lock.
+func TestCommitSweepResolverDelay(t *testing.T) {
+	t.Parallel()
+	defer timeout(t)()
+
+	res := lnwallet.CommitOutputResolution{
+		SelfOutputSignDesc: input.SignDescriptor{
+			Output: &wire.TxOut{
+				Value: 100,
+			},
+		},
+		MaturityDelay: 3,
+	}
+
+	ctx := newCommitSweepResolverTestContext(t, &res)
```

Comment:
check the report at various times during the test?

Reply:
check added